File: java/junit/basic/BasicTestSetup.java

public class BasicTestSetup {

	private AppiumDriver driver;

	@Before
	public void setUp() throws Exception {

		DesiredCapabilities capabilities = new DesiredCapabilities();

		/*  1. Choose your project */
		capabilities.setCapability("testobject_api_key", "YOUR_API_KEY");

		/*  2. Select your testing device:
		*   Specify a `platformVersion` without `deviceName` to get any available device with that platform version.
		*   Not selecting a device at all is also a valid choice, we will select it carefully for you */
		capabilities.setCapability("platformVersion", "7"); // Optional
		capabilities.setCapability("deviceName", "Samsung Galaxy S8"); // Optional

		/*  3. Where is your device located? */
		URL EU_endpoint = new URL("https://eu1.appium.testobject.com/wd/hub");
		URL US_endpoint = new URL("https://us1.appium.testobject.com/wd/hub");

		/*  The driver will take care of establishing the connection, so we must provide
		*  it with the correct endpoint and the requested capabilities. */
		driver = new AndroidDriver(EU_endpoint, capabilities);

	}

	/* We disable the driver after EACH test has been executed. */
	@After
	public void tearDown() {
		driver.quit();
	}

	@Test
	public void testMethod() {
		/* Your test. */
	}

}

File: ruby/test_unit/basic/BasicTestSetup.rb
require 'appium_lib'
require 'test/unit'

class BasicTestSetup < Test::Unit::TestCase
  API_KEY = 'YOUR_API_KEY'

  def api_key
    if API_KEY == 'YOUR_API_KEY'
      raise "Please set your API key to run this example (see https://github.com/testobject-sample-scripts/setups/tree/master/ruby/test_unit/watcher)"
      exit 1
    else
      return API_KEY
    end
  end

  def setup
    desired_capabilities = {
        caps: {
            testobject_api_key: api_key,
            platformName: 'Android',
            # supply `platformVersion` without `deviceName` to get any available device with that version dynamically
            platformVersion: '7', # Optional
            deviceName: 'Samsung Galaxy S8' # Optional
        },
        appium_lib: {
            server_url: 'https://eu1.appium.testobject.com/wd/hub',
            #server_url: 'https://us1.appium.testobject.com/wd/hub',  # your device is located in the US ? , then uncomment this line and remove the previous one.
            wait: 10
        }
    }

    # Start the driver
    @driver = Appium::Driver.new(desired_capabilities).start_driver
  end

  # test method names must start with "test_"
  # to be recognized by the test-unit framework
  def test_login
    # Your test
  end

  def teardown
    @driver.quit if @driver
  rescue Selenium::WebDriver::Error::UnknownError => e
    raise e unless e.message.match /Unable to find session with requested ID/
  end
end
